File: tests/cases/java/comment-in-header/src/pkg/Scanner.java
package pkg;

public class Scanner {
    public String scan(String root, int depth) {
        return root + depth;
    }

    public String resolve(String name) {
        return name;
    }

    public static void main(String[] args) {
        Scanner s = new Scanner();
        s.scan(".", 1);
        s.resolve("x");
    }
}

File: tests/cases/typescript/comment-in-header/src/scanner.ts
function Route(path: string) { return (t: any, k: string) => {}; }

export class Scanner {
  public async scanForModules(
    root: string,
    depth: number,
  ): Promise<string[]> {
    return [root, String(depth)];
  }

  public resolve(name: string): string {
    return name;
  }

  @Route('/a/:id')
  public handle(id: string): string {
    return id;
  }

  public scanUrl(url: string = 'http://x/*'): string {
    return url;
  }
}

export function run(s: Scanner) {
  s.scanForModules('.', 1);
  s.resolve('x');
  s.handle('1');
  s.scanUrl();
}
